media_driver/cmake/linux/media_gen_flags_linux.cmake

```diff
@@ -18,6 +18,8 @@
 # ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
 # OTHER DEALINGS IN THE SOFTWARE.
 
+option(BROKEN "Accept potentially broken builds when disabling generation-specific options" OFF)
```

Comment:
BROKEN=ON, then could enable per-gen build. what's the motivation about this build option, why GENXX = OFF is not enough?

Reply:
Currently the per-Gen build options are broken, except for Gen8.

But the per-Gen build options control two separate things: 1. The per-Gen media kernel binaries, and 2. The per-Gen driver code.

The per-Gen driver code cannot be disabled arbitrarily now.

The per-Gen media kernel binaries can be disabled arbitrarily, and they are large and disabling them provides large file size reductions.

The BROKEN option controls whether the per-Gen driver code is affected by the per-Gen config options, allowing us to disable the per-Gen media kernel binaries.

Reply:
I understand the desire to have options to reduce driver footprint, but it's not good idea to workaround something which should actually be fixed. Issue with gen10 switch off is a clear issue and needs proper resolution.

I think we should not introduce BROKEN option. Instead, I suggest to start introducing new build configurations (w/ disabled gens) which currently works to make sure we don't regress. For example, -DGEN8=OFF got fixed, let's have this path covered in github actions ci. And add other configs which currently build ok.

Reply:
I have some work in progress patches to fix Gen9, but I don't expect to be able to fix disabling the driver code for Gen10, Gen11, or Gen12.

Is someone from Intel going to sign up for that?

We can of course ship my patch as is in ChromeOS to be able to reduce the binary size, but I feel like I've provided a solid and incremental path forward that gets 90% of the size reduction, strictly improves the situation, and doesn't force anyone to drop what they're doing to untangle the currently broken per-Gen driver code disablement.

Alternatively, I'm happy to either
 -  rename the `BROKEN` option if the name is the problem
 - or I can remove `BROKEN` entirely and make the per-Gen driver code that is currently required non-optional, leaving the `GEN*` options to only disable the media kernel binaries. I create the `BROKEN` option to make the incremental work of fixing the remaining Gen options simpler.

Reply:
> * rename the BROKEN option if the name is the problem

Having received no feedback for a month, I've speculatively made this change, changing the option name to `-DONLY_MEDIA_KERNELS` and inverting the value (so that you have to set it to `OFF` to produce a broken build with `-DGEN*=OFF`).